### ISSUE TITLE
google_maps_flutter: Restrict zoom level visibility of map features

### ISSUE THREAD

**Issue by jbxbergdev** [flutter/flutter#50448]
## Use case
I'd like to show polygons/markers/lines only within a specific zoom level range, e.g. when zoom is greater than 18.0.

## Proposal
Add fields to map features minZoom/maxZoom.

**Comment by JamesMcIntosh** [comment]
You can do this behaviour already in dart

1. Make your widget statefull
2. Use `onCameraMove` to record the CameraPosition and use setState to trigger an update if the zoom level changes
3. When you build the GoogleMap in your widget just filter in/out the polygons

**Comment by darshankawar** [comment]
@jbxbergdev 
Did you give above solution a try to see if it serves your requirement ?
Thanks.

**Comment by jbxbergdev** [comment]
> @jbxbergdev 
> Did you give above solution a try to see if it serves your requirement ?
> Thanks.
> 

No, I didn't. I'd prefer if this to be handled within the Google Maps plugin.

**Comment by stuartmorgan-g** [comment]
This is not a feature that exists in the underlying SDKs, and the goal of the plugin is to provide access to the SDK functionality, so closing as out of scope.
